src/hotspot/share/code/codeBlob.hpp

```diff
@@ -318,11 +333,19 @@ class BufferBlob: public RuntimeBlob {
 
   static void free(BufferBlob* buf);
 
-  // Verification support
-  void verify() override;
+  void print_on(outputStream* st) const;
+  void print_value_on(outputStream* st) const;
 
-  void print_on(outputStream* st) const override;
-  void print_value_on(outputStream* st) const override;
+  class Vptr : public CodeBlob::Vptr {
```

Comment:
I wonder if these should share the same type hierarchy as tier container class. This would also solve the issueI noted in my other comment about not calling the correct `print_on`.
```suggestion
  class Vptr : public RuntimeBlob::Vptr {
```